modules/balana-core/src/main/java/org/wso2/balana/ctx/BasicEvaluationCtx.java

```diff
@@ -57,11 +57,6 @@
     protected Node requestRoot;
 
     /**
-     * Represents a XACML request made to the PDP
-     */
-    protected AbstractRequestCtx requestCtx;
```

Comment:
requestCtx is redeclared with a different type in XACML2EvaluationCtx and XACML3EvaluationCtx. For objects of those types there would be 2 properties with the same name so during getRequestCtx() it would return null even though one of them was set.